packages/prices-api/src/index.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Options for making API requests
+ * @example
+ * const options: Options = {
+ *   host: Promise.resolve("http://localhost:3000"), // Custom host for local testing
+ *   signal: new AbortController().signal // For request cancellation
+ * }
+ */
+export type Options = {
+  host?: Promise<string>
+  signal?: AbortSignal
+}
+
+/**
+ * Gets the API host URL, using the provided host or falling back to default
+ * @param options - Optional configuration options
+ * @returns Promise that resolves to the host URL
+ * @example
+ * const host = await getHost() // "https://prices.curve.fi"
+ */
+export const getHost = (options?: Options): Required<Options>['host'] =>
```

Comment:
Why would `getHost` need to be async :thinking:

Reply:
Good call; a leftover from before the refactor from class to functions-only. Property and function are now made non-async. I keep the `const host = getHost(options)` for readability in the api functions. When there's more query parameters being used inlining becomes a bit unreadable.